compiler/stable_mir/src/mir/body.rs

```diff
@@ -475,6 +501,63 @@ pub enum Rvalue {
     Use(Operand),
 }
 
+impl Rvalue {
+    pub fn ty(&self, locals: &[LocalDecl]) -> Result<Ty, Error> {
```

Comment:
It's a bit sad that we have to replicate this logic by hand when it exists in the compiler already, especially when it's hard to detect when `Rvalue::ty` here is returning a wrong answer (e.g. simd on binops)